tests/test_applications.py

```diff
@@ -81,7 +81,8 @@ async def websocket_raise_custom(websocket: WebSocket):
     raise CustomWSException()
 
 
-def custom_ws_exception_handler(websocket: WebSocket, exc: CustomWSException):
+def custom_ws_exception_handler(websocket: WebSocket, exc: Exception) -> None:
```

Comment:
Was there an issue with the typing on `exc`?

Reply:
The type was not being checked.

Reply:
what was the mypy issue?

Reply:
from memory...  something like `item 4 in the dict does not match the specification.`.

Look, if you specify a callable which takes a certain type as an argument, you cannot provide a callable which expects a _narrower_ type.  

This here is invalid typing:
```python
from typing import Callable
def set_callback(callback: Callable[[Exception], None]) -> None:
    pass
    
def my_callback(arg: RuntimeError)->None:
    pass
    
set_callback(my_callback)  # errors
```
running mypy on this results in:
```
test.py:8: error: Argument 1 to "set_callback" has incompatible type "Callable[[RuntimeError], None]"; expected "Callable[[Exception], None]"  [arg-type]
```